### PR TITLE
DigraphFromDigraph6String ability to return mutable digraphs

### PR DESCRIPTION
**Pull request by MTWhyte**
This pull request adds the functionality for `DigraphFromDigraph6String` to return mutable digraphs.

### CODE REVIEW

**Comment by james-d-mitchell** [comment]
Looks good! What about the other functions in the same file??

File: gap/io.gi [hunk 1]
```diff
@@ -761,17 +762,28 @@ function(func, s)
...
 
   if legacy then  # source and range are reversed
-    return func(rec(DigraphNrVertices := n,
-                    DigraphSource     := range,
-                    DigraphRange      := source));
-  fi;
-  return func(rec(DigraphNrVertices := n,
-                  DigraphRange      := range,
-                  DigraphSource     := source));
+    return DigraphNC(IsMutableDigraph,
```
Comment (wilfwilson) [inline gap/io.gi]: I don't have any idea why this code was originally written like this, but you should check with at least @james-d-mitchell that we're losing anything important by removing the ability of `DIGRAPHS_FromDigraph6String` to take an arbitrary "constructor", and forcing `DigraphNC`.

File: gap/io.gi [hunk 2]
```diff
@@ -761,17 +762,28 @@ function(func, s)
...
 end);
 
+InstallMethod(DigraphFromDigraph6StringCons,
+"for IsImmutableDigraph and a string",
+[IsImmutableDigraph, IsString],
+{filt, s} -> MakeImmutable(DigraphFromDigraph6StringCons(IsMutableDigraph, s)));
+
+InstallMethod(DigraphFromDigraph6String, "for a function and a string",
+[IsFunction, IsString],
+{func, s} -> DigraphFromDigraph6StringCons(func, s));
```
Comment (wilfwilson) [inline gap/io.gi]:
```suggestion
DigraphFromDigraph6StringCons);
```

File: gap/io.gi (hunk 1 above)
Comment (MTWhyte) [reply gap/io.gi]: Yeah, we're definitely ok to do this - getting rid of `DIGRAPHS_FromDigraph6String` was actually James's suggestion.

**Comment by MTWhyte** [comment]
Great, I'll change the others too!

**Comment by wilfwilson** [comment]
Great, thanks a lot. Really handy.

File: gap/io.gi (hunk 1 above)
Comment (james-d-mitchell) [reply gap/io.gi]: I concur, it was my idea, I think this is the best solution.

File: gap/io.gi (hunk 2 above)
Comment (james-d-mitchell) [reply gap/io.gi]: Nice

**Comment by wilfwilson** [comment]
@MTWhyte do you mean you'll do it in this PR, or in a separate PR?

**Comment by MTWhyte** [comment]
@wilfwilson which do you think would be better?

**Comment by wilfwilson** [comment]
I think you may as well do it all in this PR, but I don’t mind if you want to do it separately.

**Comment by MTWhyte** [comment]
I've now updated the 'friends' too, namely `DigraphFromGraph6String`, `DigraphFromSparse6String`, `DigraphFromDiSparse6String` and `DigraphFromPlainTextString`.

**Pull request merged**